docs/en/writing-running-appium/image-comparison.md

```diff
@@ -59,6 +59,67 @@ File.open('match_result_visual.png', 'wb') { |f| f<< Base64.decode64(match_resul
 assert File.size? 'match_result_visual.png'
 ```
 
+```javascript
+Typescript / Javascript
+  /*
+     Typescsript code for occurence comparison using the template matching algorithm.
+     It detects if an image is contained in another image (called the template).
+
+     There is a subtle difference between occurrence comparison and feature comparison. The former is to be used when the image to be found is a subset of the target/screenshot; the latter is to be used when the image to be found is basically the same as the target but rotated and/or scaled.
+
+     official doc:
+     https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/image-comparison.md
+     OpenCV algorithm doc:
+     https://docs.opencv.org/2.4/doc/tutorials/imgproc/histograms/template_matching/template_matching.html
+     official sample code:
+     https://github.com/justadudewhohacks/opencv4nodejs/blob/master/examples/templateMatching.js
+
+     You must install opencv4nodejs using the -g option.
+
+     The Javascript client does not support (in January 2020) the "-image" strategy implemented in the Appium server. You will have more power and understanding while using openCV directly. Since The appium server is in Javascript, you can do all it does with opencv in the client.
+
+     The testing framework mocha can be run with typescript to have async/await.
+     You need to run mocha with those options in the right order and the associated packages:
+     TS_NODE_PROJECT=config/tsconfig_test.json --require ts-node/register --require tsconfig-paths/register
+     You will also need to make a basic config/tsconfig_test.json and you will need an environemnt
+     variable giving the absolute path to where opencv4nodejs is located in nodejs
+  */
+  const cv = require(process.env.OPENCV4NODEJS_PATH + '/lib/node_modules/opencv4nodejs');
+  const isImagePresent = async () => {
+    /// Take screenshot and read the image
+    const screenImagePath = './appium_screenshot1.png';
+    driver.saveScreenshot(screenImagePath)
+    const screenImage = cv.imread(screenImagePath);
```

Comment:
I'd prefer async functions to sync if such functions exist

Reply:
fixed